File: LearnerBee/src/screens/InstructorDetailScreen.js
// src/screens/InstructorDetailScreen.js
import React, { useState, useEffect } from 'react';
import {
  View,
  Text,
  StyleSheet,
  Image,
  ScrollView,
  TouchableOpacity,
  SafeAreaView,
} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';
import { instructors, toggleFavorite } from '../services/mockData';

const InstructorDetailScreen = ({ route }) => {
  const { instructorId } = route.params;
  const [instructor, setInstructor] = useState(null);
  const [isFavorite, setIsFavorite] = useState(false);

  useEffect(() => {
    // Get instructor details from mock data
    const foundInstructor = instructors.find(i => i.id === instructorId);
    if (foundInstructor) {
      setInstructor(foundInstructor);
      setIsFavorite(foundInstructor.isFavorite);
    }
  }, [instructorId]);

  const handleFavoritePress = () => {
    const newState = toggleFavorite(instructorId);
    setIsFavorite(newState);
  };

  if (!instructor) {
    return (
      <View style={styles.loadingContainer}>
        <Text>Loading instructor details...</Text>
      </View>
    );
  }

  return (
    <SafeAreaView style={styles.container}>
      <ScrollView>
        <View style={styles.profileHeader}>
          <Image source={{ uri: instructor.image }} style={styles.profileImage} />
          
          <View style={styles.nameContainer}>
            <Text style={styles.name}>{instructor.name}</Text>
            <Text style={styles.location}>{instructor.location}</Text>
            
            <View style={styles.ratingContainer}>
              <Text style={styles.ratingNumber}>{instructor.rating}</Text>
              <Icon name="star" size={16} color="#FFD700" />
              <Text style={styles.reviewCount}>({instructor.reviewCount} reviews)</Text>
            </View>
          </View>
          
          <TouchableOpacity 
            style={styles.favoriteButton}
            onPress={handleFavoritePress}
          >
            <Icon 
              name={isFavorite ? "heart" : "heart-o"} 
              size={24} 
              color={isFavorite ? "#4A6FF3" : "#8E8E8E"} 
            />
          </TouchableOpacity>
        </View>
        
        <View style={styles.infoSection}>
          <Text style={styles.sectionTitle}>About</Text>
          <Text style={styles.experienceText}>
            {instructor.experience}
          </Text>
        </View>
        
        <View style={styles.infoSection}>
          <Text style={styles.sectionTitle}>Lesson Details</Text>
          
          <View style={styles.detailRow}>
            <Icon name="clock-o" size={20} color="#666" style={styles.detailIcon} />
            <Text style={styles.detailText}>
              {instructor.duration} lessons
            </Text>
          </View>
          
          <View style={styles.detailRow}>
            <Icon name="credit-card" size={20} color="#666" style={styles.detailIcon} />
            <Text style={styles.detailText}>
              ${instructor.price} per lesson
            </Text>
          </View>
          
          <View style={styles.detailRow}>
            <Icon name="map-marker" size={20} color="#666" style={styles.detailIcon} />
            <Text style={styles.detailText}>
              Serves {instructor.location} area
            </Text>
          </View>
        </View>
        
        <View style={styles.reviewsSection}>
          <Text style={styles.sectionTitle}>Reviews</Text>
          
          {/* Placeholder for reviews - would be dynamic in a real app */}
          <View style={styles.reviewItem}>
            <View style={styles.reviewHeader}>
              <Text style={styles.reviewName}>John D.</Text>
              <View style={styles.reviewRating}>
                <Icon name="star" size={14} color="#FFD700" />
                <Icon name="star" size={14} color="#FFD700" />
                <Icon name="star" size={14} color="#FFD700" />
                <Icon name="star" size={14} color="#FFD700" />
                <Icon name="star" size={14} color="#FFD700" />
              </View>
            </View>
            <Text style={styles.reviewText}>
              Amazing instructor! Very patient and great at explaining complex maneuvers.
              Helped me pass my test on the first try.
            </Text>
          </View>
          
          <View style={styles.reviewItem}>
            <View style={styles.reviewHeader}>
              <Text style={styles.reviewName}>Sarah T.</Text>
              <View style={styles.reviewRating}>
                <Icon name="star" size={14} color="#FFD700" />
                <Icon name="star" size={14} color="#FFD700" />
                <Icon name="star" size={14} color="#FFD700" />
                <Icon name="star" size={14} color="#FFD700" />
                <Icon name="star" size={14} color="#FFD700" />
              </View>
            </View>
            <Text style={styles.reviewText}>
              Extremely professional and knowledgeable. Made me feel comfortable
              behind the wheel right from the start.
            </Text>
          </View>
        </View>
      </ScrollView>
      
      <View style={styles.bottomBar}>
        <View style={styles.priceContainer}>
          <Text style={styles.priceLabel}>Price</Text>
          <Text style={styles.price}>${instructor.price}</Text>
          <Text style={styles.pricePer}>per {instructor.duration} lesson</Text>
        </View>
        
        <TouchableOpacity style={styles.bookButton}>
          <Text style={styles.bookButtonText}>Book Lesson</Text>
        </TouchableOpacity>
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#F8F8F8',
  },
  loadingContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  profileHeader: {
    flexDirection: 'row',
    padding: 16,
    backgroundColor: 'white',
    borderBottomWidth: 1,
    borderBottomColor: '#EEEEEE',
  },
  profileImage: {
    width: 80,
    height: 80,
    borderRadius: 40,
  },
  nameContainer: {
    flex: 1,
    marginLeft: 16,
    justifyContent: 'center',
  },
  name: {
    fontSize: 22,
    fontWeight: 'bold',
    marginBottom: 4,
  },
  location: {
    fontSize: 16,
    color: '#666',
    marginBottom: 4,
  },
  ratingContainer: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  ratingNumber: {
    fontSize: 16,
    fontWeight: 'bold',
    marginRight: 4,
  },
  reviewCount: {
    fontSize: 14,
    color: '#666',
    marginLeft: 4,
  },
  favoriteButton: {
    padding: 8,
  },
  infoSection: {
    padding: 16,
    backgroundColor: 'white',
    marginTop: 12,
  },
  sectionTitle: {
    fontSize: 18,
    fontWeight: 'bold',
    marginBottom: 12,
  },
  experienceText: {
    fontSize: 16,
    lineHeight: 24,
    color: '#333',
  },
  detailRow: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 12,
  },
  detailIcon: {
    width: 24,
    marginRight: 12,
  },
  detailText: {
    fontSize: 16,
    color: '#333',
  },
  reviewsSection: {
    padding: 16,
    backgroundColor: 'white',
    marginTop: 12,
    marginBottom: 80, // Add space for the bottom bar
  },
  reviewItem: {
    paddingVertical: 12,
    borderBottomWidth: 1,
    borderBottomColor: '#EEEEEE',
  },
  reviewHeader: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginBottom: 8,
  },
  reviewName: {
    fontSize: 16,
    fontWeight: 'bold',
  },
  reviewRating: {
    flexDirection: 'row',
  },
  reviewText: {
    fontSize: 14,
    lineHeight: 20,
    color: '#333',
  },
  bottomBar: {
    position: 'absolute',
    bottom: 0,
    left: 0,
    right: 0,
    flexDirection: 'row',
    padding: 16,
    backgroundColor: 'white',
    borderTopWidth: 1,
    borderTopColor: '#EEEEEE',
    alignItems: 'center',
  },
  priceContainer: {
    flex: 1,
  },
  priceLabel: {
    fontSize: 12,
    color: '#666',
  },
  price: {
    fontSize: 20,
    fontWeight: 'bold',
  },
  pricePer: {
    fontSize: 12,
    color: '#666',
  },
  bookButton: {
    backgroundColor: '#4A6FF3',
    paddingVertical: 12,
    paddingHorizontal: 24,
    borderRadius: 8,
  },
  bookButtonText: {
    color: 'white',
    fontSize: 16,
    fontWeight: 'bold',
  },
});

export default InstructorDetailScreen;

File: LearnerBee/lib/screens/instructor_detail_screen.dart
import 'package:flutter/material.dart';
import '../models/instructor.dart';
import '../services/instructor_service.dart';
import 'payment_screen.dart';

class InstructorDetailScreen extends StatefulWidget {
  final String instructorId;
  
  InstructorDetailScreen({required this.instructorId});
  
  @override
  _InstructorDetailScreenState createState() => _InstructorDetailScreenState();
}

class _InstructorDetailScreenState extends State<InstructorDetailScreen> {
  late Instructor instructor;
  late bool isFavorite;
  
  @override
  void initState() {
    super.initState();
    instructor = InstructorService.getInstructorById(widget.instructorId);
    isFavorite = instructor.isFavorite;
  }
  
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Instructor Profile'),
      ),
      body: Column(
        children: [
          // Instructor profile header
          Container(
            padding: EdgeInsets.all(16),
            color: Colors.white,
            child: Row(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                // Profile image
                ClipRRect(
                  borderRadius: BorderRadius.circular(40),
                  child: Image.network(
                    instructor.image,
                    width: 80,
                    height: 80,
                    fit: BoxFit.cover,
                  ),
                ),
                SizedBox(width: 16),
                
                // Instructor info
                Expanded(
                  child: Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: [
                      Text(
                        instructor.name,
                        style: TextStyle(
                          fontSize: 22,
                          fontWeight: FontWeight.bold,
                        ),
                      ),
                      SizedBox(height: 4),
                      Text(
                        instructor.location,
                        style: TextStyle(
                          fontSize: 16,
                          color: Colors.grey[600],
                        ),
                      ),
                      SizedBox(height: 4),
                      Row(
                        children: [
                          Text(
                            instructor.rating.toString(),
                            style: TextStyle(
                              fontSize: 16,
                              fontWeight: FontWeight.bold,
                            ),
                          ),
                          SizedBox(width: 4),
                          Icon(Icons.star, color: Colors.amber, size: 16),
                          SizedBox(width: 4),
                          Text(
                            '(${instructor.reviewCount} reviews)',
                            style: TextStyle(
                              fontSize: 14,
                              color: Colors.grey[600],
                            ),
                          ),
                        ],
                      ),
                    ],
                  ),
                ),
                
                // Favorite button
                IconButton(
                  icon: Icon(
                    isFavorite ? Icons.favorite : Icons.favorite_border,
                    color: isFavorite ? Theme.of(context).primaryColor : Colors.grey,
                  ),
                  onPressed: () {
                    setState(() {
                      InstructorService.toggleFavorite(instructor.id);
                      isFavorite = !isFavorite;
                    });
                  },
                ),
              ],
            ),
          ),
          
          // About section
          Expanded(
            child: ListView(
              children: [
                // About section
                _buildSection(
                  title: 'About',
                  child: Text(
                    instructor.experience,
                    style: TextStyle(
                      fontSize: 16,
                      color: Colors.black87,
                      height: 1.5,
                    ),
                  ),
                ),
                
                // Lesson details section
                _buildSection(
                  title: 'Lesson Details',
                  child: Column(
                    children: [
                      _buildDetailRow(Icons.access_time, '${instructor.duration} lessons'),
                      SizedBox(height: 12),
                      _buildDetailRow(Icons.credit_card, '\$${instructor.price.toInt()} per lesson'),
                      SizedBox(height: 12),
                      _buildDetailRow(Icons.location_on, 'Serves ${instructor.location} area'),
                    ],
                  ),
                ),
                
                // Reviews section
                _buildSection(
                  title: 'Reviews',
                  child: Column(
                    children: [
                      // Mock reviews - in a real app, these would come from a service
                      _buildReview(
                        name: 'John D.',
                        rating: 5,
                        comment: 'Amazing instructor! Very patient and great at explaining complex maneuvers. Helped me pass my test on the first try.',
                      ),
                      SizedBox(height: 16),
                      _buildReview(
                        name: 'Sarah T.',
                        rating: 5,
                        comment: 'Extremely professional and knowledgeable. Made me feel comfortable behind the wheel right from the start.',
                      ),
                    ],
                  ),
                ),
                
                // Add space for bottom bar
                SizedBox(height: 80),
              ],
            ),
          ),
        ],
      ),
      
      // Bottom bar with price and book button
      bottomSheet: Container(
        height: 80,
        padding: EdgeInsets.all(16),
        decoration: BoxDecoration(
          color: Colors.white,
          boxShadow: [
            BoxShadow(
              color: Colors.black.withOpacity(0.05),
              blurRadius: 5,
              offset: Offset(0, -1),
            ),
          ],
        ),
        child: Row(
          children: [
            // Price
            Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              mainAxisSize: MainAxisSize.min,
              children: [
                Text(
                  'Price',
                  style: TextStyle(
                    fontSize: 12,
                    color: Colors.grey[600],
                  ),
                ),
                Text(
                  '\$${instructor.price.toInt()}',
                  style: TextStyle(
                    fontSize: 20,
                    fontWeight: FontWeight.bold,
                  ),
                ),
                Text(
                  'per ${instructor.duration} lesson',
                  style: TextStyle(
                    fontSize: 12,
                    color: Colors.grey[600],
                  ),
                ),
              ],
            ),
            SizedBox(width: 16),
            
            // Book button
            Expanded(
              child: ElevatedButton(
                onPressed: () {
                  Navigator.push(
                    context,
                    MaterialPageRoute(
                      builder: (context) => PaymentScreen(
                        instructor: instructor,
                        lessonDuration: instructor.duration,
                        price: instructor.price,
                      ),
                    ),
                  );
                  // Book lesson functionality would go here
                  // ScaffoldMessenger.of(context).showSnackBar(
                  //   SnackBar(content: Text('Booking feature coming soon!')),
                  // );
                },
                style: ElevatedButton.styleFrom(
                  padding: EdgeInsets.symmetric(vertical: 12),
                  shape: RoundedRectangleBorder(
                    borderRadius: BorderRadius.circular(8),
                  ),
                ),
                child: Text(
                  'Book Lesson',
                  style: TextStyle(
                    fontWeight: FontWeight.bold,
                    fontSize: 16,
                  ),
                ),
              ),
            ),
          ],
        ),
      ),
    );
  }
  
  Widget _buildSection({required String title, required Widget child}) {
    return Container(
      margin: EdgeInsets.only(top: 12),
      padding: EdgeInsets.all(16),
      color: Colors.white,
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Text(
            title,
            style: TextStyle(
              fontSize: 18,
              fontWeight: FontWeight.bold,
            ),
          ),
          SizedBox(height: 12),
          child,
        ],
      ),
    );
  }
  
  Widget _buildDetailRow(IconData icon, String text) {
    return Row(
      children: [
        Icon(
          icon,
          color: Colors.grey[600],
          size: 20,
        ),
        SizedBox(width: 12),
        Text(
          text,
          style: TextStyle(
            fontSize: 16,
            color: Colors.black87,
          ),
        ),
      ],
    );
  }
  
  Widget _buildReview({
    required String name,
    required int rating,
    required String comment,
  }) {
    return Container(
      padding: EdgeInsets.symmetric(vertical: 12),
      decoration: BoxDecoration(
        border: Border(
          bottom: BorderSide(color: Colors.grey[300]!),
        ),
      ),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Row(
            mainAxisAlignment: MainAxisAlignment.spaceBetween,
            children: [
              Text(
                name,
                style: TextStyle(
                  fontSize: 16,
                  fontWeight: FontWeight.bold,
                ),
              ),
              Row(
                children: List.generate(5, (index) {
                  return Icon(
                    Icons.star,
                    size: 14,
                    color: index < rating ? Colors.amber : Colors.grey[300],
                  );
                }),
              ),
            ],
          ),
          SizedBox(height: 8),
          Text(
            comment,
            style: TextStyle(
              fontSize: 14,
              color: Colors.black87,
              height: 1.5,
            ),
          ),
        ],
      ),
    );
  }
}